src/tools/validation.rs

```diff
@@ -50,7 +50,7 @@ pub fn validate_blob(path: &Path) -> AnyResult<()> {
     let mut reader = BlobReader::from_path(&path)?;
     reader.read_header()?;
     while !reader.is_eof() {
-        reader.read_record(false)?;
+        reader.read_record(true)?;
```

Comment:
You should fix `read_record` instead. It looks like `if skip_wrong` condition at the beggining of `read_record` should be inversed. Additionally, validate for correctness all places in Pearl and Bob, where `read_record` is called